pkg/conversions/virtualnode_types_conversion.go

```diff
@@ -0,0 +1,329 @@
+	sdkObj.NamespaceName = aws.String(crdObj.NamespaceName)
+	sdkObj.ServiceName = aws.String(crdObj.ServiceName)
+
+	var sdkAttributes []*appmeshsdk.AwsCloudMapInstanceAttribute
+	if len(crdObj.Attributes) != 0 {
+		sdkAttributes = make([]*appmeshsdk.AwsCloudMapInstanceAttribute, 0, len(crdObj.Attributes))
+		for _, crdAttribute := range crdObj.Attributes {
+			sdkAttribute := &appmeshsdk.AwsCloudMapInstanceAttribute{}
+			if err := Convert_CRD_AWSCloudMapInstanceAttribute_To_SDK_AWSCloudMapInstanceAttribute(&crdAttribute, sdkAttribute, scope); err != nil {
+				return err
+			}
+			sdkAttributes = append(sdkAttributes, sdkAttribute)
+		}
+	}
+	sdkObj.Attributes = sdkAttributes
+	return nil
+}
+
+func Convert_CRD_DNSServiceDiscovery_To_SDK_DNSServiceDiscovery(crdObj *appmesh.DNSServiceDiscovery, sdkObj *appmeshsdk.DnsServiceDiscovery, scope conversion.Scope) error {
+	sdkObj.Hostname = aws.String(crdObj.Hostname)
+	return nil
+}
+
+func Convert_CRD_ServiceDiscovery_To_SDK_ServiceDiscovery(crdObj *appmesh.ServiceDiscovery, sdkObj *appmeshsdk.ServiceDiscovery, scope conversion.Scope) error {
+	if crdObj.AWSCloudMap != nil {
+		sdkObj.AwsCloudMap = &appmeshsdk.AwsCloudMapServiceDiscovery{}
+		if err := Convert_CRD_AWSCloudMapServiceDiscovery_To_SDK_AWSCloudMapServiceDiscovery(crdObj.AWSCloudMap, sdkObj.AwsCloudMap, scope); err != nil {
+			return err
+		}
+	} else {
+		sdkObj.AwsCloudMap = nil
+	}
+
+	if crdObj.DNS != nil {
+		sdkObj.Dns = &appmeshsdk.DnsServiceDiscovery{}
+		if err := Convert_CRD_DNSServiceDiscovery_To_SDK_DNSServiceDiscovery(crdObj.DNS, sdkObj.Dns, scope); err != nil {
+			return err
+		}
+	} else {
+		sdkObj.Dns = nil
+	}
+
+	return nil
+}
+
+func Convert_CRD_FileAccessLog_To_SDK_FileAccessLog(crdObj *appmesh.FileAccessLog, sdkObj *appmeshsdk.FileAccessLog, scope conversion.Scope) error {
+	sdkObj.Path = aws.String(crdObj.Path)
+	return nil
+}
+
+func Convert_CRD_AccessLog_To_SDK_AccessLog(crdObj *appmesh.AccessLog, sdkObj *appmeshsdk.AccessLog, scope conversion.Scope) error {
+	if crdObj.File != nil {
+		sdkObj.File = &appmeshsdk.FileAccessLog{}
+		if err := Convert_CRD_FileAccessLog_To_SDK_FileAccessLog(crdObj.File, sdkObj.File, scope); err != nil {
+			return err
+		}
+	} else {
+		sdkObj.File = nil
+	}
+	return nil
+}
+
+func Convert_CRD_Logging_To_SDK_Logging(crdObj *appmesh.Logging, sdkObj *appmeshsdk.Logging, scope conversion.Scope) error {
+	if crdObj.AccessLog != nil {
+		sdkObj.AccessLog = &appmeshsdk.AccessLog{}
+		if err := Convert_CRD_AccessLog_To_SDK_AccessLog(crdObj.AccessLog, sdkObj.AccessLog, scope); err != nil {
+			return err
+		}
+	} else {
+		sdkObj.AccessLog = nil
+	}
+	return nil
+}
+
+func Convert_CRD_VirtualNodeSpec_To_SDK_VirtualNodeSpec(crdObj *appmesh.VirtualNodeSpec, sdkObj *appmeshsdk.VirtualNodeSpec, scope conversion.Scope) error {
```

Comment:
It'd be good to add testing (with 0, 1, >1 listeners, backends) for `Convert_CRD_VirtualNodeSpec_To_SDK_VirtualNodeSpec` as well

Reply:
sure